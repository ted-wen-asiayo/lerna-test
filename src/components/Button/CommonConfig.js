import styled from 'styled-components'
import {
  variant,
  color,
  space,
  shadow,
  border,
  typography,
} from 'styled-system'

const commonStyled = (component) => styled(component)`
  &[type] {
    box-sizing: border-box;
    display: flex;
    align-items: center;
    justify-content: center;
    cursor: pointer;
    &:disabled {
      opacity: ${(props) => (props.loading ? '1' : '0.48')};
      cursor: not-allowed;
    }
    ${color}
    ${space}
    ${shadow}
    ${border}
    ${typography}
  }
`
const size = variant({
  prop: 'size',
  variants: {
    small: {
      fontSize: '14px',
      lineHeight: '20px',
      fontWeight: '500',
      padding: '6px 16px',
      borderWidth: '1px',
      borderStyle: 'solid',
    },
    medium: {
      fontSize: '16px',
      lineHeight: '24px',
      fontWeight: '500',
      padding: '8px 20px',
      borderWidth: '1px',
      borderStyle: 'solid',
    },
    large: {
      fontSize: '16px',
      lineHeight: '24px',
      fontWeight: '500',
      padding: '12px 24px',
      borderWidth: '1px',
      borderStyle: 'solid',
    },
    response: {
      fontSize: '16px',
      lineHeight: '24px',
      fontWeight: '500',
      padding: ['12px 24px', null, '12px 24px', null, '8px 20px'],
      borderWidth: '1px',
      borderStyle: 'solid',
    },
  },
})

const iconSize = variant({
  prop: 'size',
  variants: {
    small: {
      fontSize: '16px',
      fontWeight: '500',
      width: '32px',
      height: '32px',
      borderWidth: '1px',
      borderStyle: 'solid',
    },
    medium: {
      fontSize: '18px',
      fontWeight: '500',
      width: '40px',
      height: '40px',
      borderWidth: '1px',
      borderStyle: 'solid',
    },
    large: {
      fontSize: '20px',
      fontWeight: '500',
      width: '48px',
      height: '48px',
      borderWidth: '1px',
      borderStyle: 'solid',
    },
    response: {
      fontSize: ['20px', null, '20px', null, '18px'],
      fontWeight: '500',
      width: ['48px', null, '48px', null, '40px'],
      height: ['48px', null, '48px', null, '40px'],
      borderWidth: '1px',
      borderStyle: 'solid',
    },
  },
})

const shape = (props) =>
  variant({
    prop: 'shape',
    variants: {
      rounded: {
        borderRadius: '4px',
      },
      pill: {
        borderRadius: props.size,
      },
    },
  })

export { commonStyled, size, iconSize, shape }
