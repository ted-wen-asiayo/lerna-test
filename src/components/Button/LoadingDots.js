import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { variant } from 'styled-system'
// import Dots, { Dot } from '@/components/Loading/Dots'

const loadingSize = props => variant({
  prop: 'size',
  variants: {
    small: {
      lineHeight: '20px',
      padding: '6px 0',
    },
    medium: {
      lineHeight: '24px',
      padding: props.isPureIcon ? '8px' : '8px 0',
    },
    large: {
      lineHeight: '24px',
      padding: props.isPureIcon ? '12px' : '8px 0',
    },
  },
})

// const StyledDots = styled(Dots)`
//   width: 44px;
//   ${Dot} {
//     border-width: ${props => (props.isPureIcon ? '2px' : '4px')};
//   }
//   ${loadingSize}
// `

const LoadingDots = ({ size, isPureIcon }) => (
  null
  // <StyledDots number={3} isPureIcon={isPureIcon} size={size} />
)

LoadingDots.propTypes = {
  isPureIcon: PropTypes.bool,
  size: PropTypes.oneOf(['small', 'medium', 'large']),
}

export default LoadingDots
