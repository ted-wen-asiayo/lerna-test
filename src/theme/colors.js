const blues = [
  "#f0fdff",
  "#c7f7ff",
  "#99e6f7",
  "#6ccfeb",
  "#43b7de",
  "#1e9fd2",
  "#0f7aab",
  "#055885",
  "#003a5e",
  "#002138",
];

const greens = [
  "#daedd5",
  "#9ce090",
  "#72d466",
  "#4bc740",
  "#26ba1e",
  "#04ad00",
  "#008702",
  "#006105",
  "#003b05",
  "#001402",
];

const yellows = [
  "#fffce6",
  "#fff0a3",
  "#ffe47a",
  "#ffd752",
  "#ffc629",
  "#ffb300",
  "#d99100",
  "#b37100",
  "#8c5400",
  "#663a00",
];

const oranges = [
  "#fff3eb",
  "#ffd8c2",
  "#ffbb99",
  "#ff9b70",
  "#ff7847",
  "#f4511e",
  "#cf350e",
  "#a81f03",
  "#821100",
  "#5c0900",
];

const grays = [
  "#ffffff",
  "#fafafa",
  "#f5f5f5",
  "#e8e8e8",
  "#d9d9d9",
  "#bfbfbf",
  "#8c8c8c",
  "#595959",
  "#262626",
  "#000000",
];

const peaches = [
  "#FFF0F4",
  "#FFD4E1",
  "#FCA9C5",
  "#F07AA5",
  "#E34F8A",
  "#D62872",
  "#B0195F",
  "#8A0C4B",
  "#630437",
  "#3D0223",
];

const hostPrimary = [
  "#E8F7F6",
  "#BCE9E4",
  "#90DAD3",
  "#63CBC2",
  "#37BDB0",
  "#21B6A8",
  "#20AEA1",
  "#1DA094",
  "#198C82",
  "#15756C",
];

const main = {
  base: blues[5],
};

export default {
  blues,
  greens,
  yellows,
  oranges,
  grays,
  peaches,
  hostPrimary,
  main,
};
