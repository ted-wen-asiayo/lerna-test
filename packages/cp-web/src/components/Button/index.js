import React from "react";
import PropTypes from "prop-types";
import { commonStyled } from "../../../../../src/components/Button/CommonConfig";
import Flat from "./Flat";
import Outline from "./Outline";
import Solid from "./Solid";
import LoadingDots from "../../../../../src/components/Button/LoadingDots";

const SwitchComponents = {
  flat: commonStyled(Flat),
  outline: commonStyled(Outline),
  solid: commonStyled(Solid),
};

const Button = ({ loading, disabled, styleType, children, ...props }) => {
  const Wrapper = SwitchComponents[styleType];

  return (
    <Wrapper disabled={loading || disabled} loading={loading} {...props}>
      {loading ? <LoadingDots {...props} /> : children}
    </Wrapper>
  );
};

Button.defaultProps = {
  type: "button",
  styleType: "solid",
  size: "small",
  shape: "rounded",
  level: "primary",
  isPureIcon: false,
};

Button.propTypes = {
  type: PropTypes.string,
  loading: PropTypes.bool,
  disabled: PropTypes.bool,
  styleType: PropTypes.oneOf(["flat", "outline", "solid"]),
  size: PropTypes.oneOf(["small", "medium", "large", "response"]),
  shape: PropTypes.oneOf(["rounded", "pill"]),
  level: PropTypes.oneOf([
    "primary",
    "success",
    "warning",
    "danger",
    "neutral",
    "tertiary",
    "dark",
    "ghost",
    "momo",
  ]),
  isPureIcon: PropTypes.bool,
};

export default Button;
