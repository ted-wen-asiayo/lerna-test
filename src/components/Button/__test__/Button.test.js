import React from 'react'
import renderer from 'react-test-renderer'
import {
  render, cleanup, fireEvent, screen,
} from '@testing-library/react'
import 'jest-styled-components'
import Icons from '../../Icons'
import colors from '../../../theme/colors'
import Button from '../index'

describe('test Button styles', () => {
  test('button shapes all render correct', () => {
    ['rounded', 'pill'].forEach((shape) => {
      const tree = renderer
        .create(<Button shape={shape}>Shape</Button>)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  test('button size all render correct', () => {
    ['small', 'medium', 'large', 'response'].forEach((size) => {
      const tree = renderer.create(<Button size={size}>Size</Button>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  const hover = '[type]:not(:disabled):hover'
  const active = '[type]:not(:disabled):active'
  const focus = '[type]:not(:disabled):focus'

  test('primary button style', () => {
    const primary = renderer
      .create(
        <Button styleType="solid" level="primary">
          Primary
        </Button>,
      )
      .toJSON()
    expect(primary).toMatchSnapshot()
    expect(primary).toHaveStyleRule('background-color', colors.blues[6], {
      modifier: hover,
    })
    expect(primary).toHaveStyleRule('background-color', colors.blues[7], {
      modifier: active,
    })
    expect(primary).toHaveStyleRule(
      'box-shadow',
      `0 0 0 2px ${colors.blues[5]}48`,
      { modifier: focus },
    )
  })

  test('secondary button style', () => {
    const secondary = renderer
      .create(
        <Button styleType="outline" level="primary">
          Secondary
        </Button>,
      )
      .toJSON()
    expect(secondary).toHaveStyleRule('background-color', colors.blues[0], {
      modifier: hover,
    })
    expect(secondary).toHaveStyleRule('background-color', colors.blues[1], {
      modifier: active,
    })
    expect(secondary).toHaveStyleRule(
      'box-shadow',
      `0 0 0 2px ${colors.blues[5]}48`,
      { modifier: focus },
    )
  })

  test('tertiary button style', () => {
    const tertiary = renderer
      .create(
        <Button styleType="outline" level="neutral">
          Tertiary
        </Button>,
      )
      .toJSON()
    expect(tertiary).toHaveStyleRule('background-color', colors.grays[2], {
      modifier: hover,
    })
    expect(tertiary).toHaveStyleRule('background-color', colors.grays[3], {
      modifier: active,
    })
    expect(tertiary).toHaveStyleRule(
      'box-shadow',
      `0 0 0 2px ${colors.grays[6]}48`,
      { modifier: focus },
    )
  })

  test('quaternary button style', () => {
    const quaternary = renderer
      .create(
        <Button styleType="flat" level="primary">
          Quaternary
        </Button>,
      )
      .toJSON()
    expect(quaternary).toHaveStyleRule('color', colors.blues[6], { modifier: hover })
    expect(quaternary).toHaveStyleRule('color', colors.blues[7], { modifier: active })
    expect(quaternary).toHaveStyleRule(
      'box-shadow',
      `0 0 0 2px ${colors.blues[5]}48`,
      { modifier: focus },
    )
  })

  test('quaternary danger button style', () => {
    const danger = renderer
      .create(
        <Button styleType="flat" level="danger">
          Delete
        </Button>,
      )
      .toJSON()
    expect(danger).toHaveStyleRule('color', colors.oranges[6], { modifier: hover })
    expect(danger).toHaveStyleRule('color', colors.oranges[7], { modifier: active })
    expect(danger).toHaveStyleRule(
      'box-shadow',
      `0 0 0 2px ${colors.oranges[5]}48`,
      { modifier: focus },
    )
  })

  test('disabled button style', () => {
    const disabled = renderer
      .create(<Button disabled>Disabled</Button>)
      .toJSON()
    expect(disabled).toMatchSnapshot()
    expect(disabled).toHaveStyleRule('opacity', '0.48', {
      modifier: '[type]:disabled',
    })
  })

  test('loading button style', () => {
    const loading = renderer.create(<Button loading>Loading</Button>).toJSON()
    expect(loading).toMatchSnapshot()
    expect(loading).toHaveStyleRule('opacity', '1', {
      modifier: '[type]:disabled',
    })
  })

  test('pureIcon button style', () => {
    const pureIconLoading = render(
      <Button isPureIcon>
        <Icons type="solid" name="home" />
        Loading
      </Button>,
    )
    expect(pureIconLoading).toMatchSnapshot()
  })
})

describe('test Button action', () => {
  afterEach(cleanup)

  test('calls onClick prop when clicked', () => {
    const handleClick = jest.fn()
    render(<Button onClick={handleClick}>Click Me</Button>)
    fireEvent.click(screen.getByText('Click Me'))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  test('calls onMouseDown prop when mouse down', () => {
    const handleMouseDown = jest.fn()
    render(<Button onMouseDown={handleMouseDown}>Press Down</Button>)
    fireEvent.mouseDown(screen.getByText('Press Down'))
    expect(handleMouseDown).toHaveBeenCalledTimes(1)
  })

  test('calls onMouseUp prop when mouse up', () => {
    const handleMouseUp = jest.fn()
    render(<Button onMouseUp={handleMouseUp}>Press Up</Button>)
    fireEvent.mouseUp(screen.getByText('Press Up'))
    expect(handleMouseUp).toHaveBeenCalledTimes(1)
  })
})
