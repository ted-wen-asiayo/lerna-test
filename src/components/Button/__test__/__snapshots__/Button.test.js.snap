// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button styles button shapes all render correct 1`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  padding: 6px 16px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  shape="rounded"
  size="small"
  type="button"
>
  Shape
</button>
`;

exports[`test Button styles button shapes all render correct 2`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: small;
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  padding: 6px 16px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {

}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  shape="pill"
  size="small"
  type="button"
>
  Shape
</button>
`;

exports[`test Button styles button size all render correct 1`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  padding: 6px 16px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  shape="rounded"
  size="small"
  type="button"
>
  Size
</button>
`;

exports[`test Button styles button size all render correct 2`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  padding: 8px 20px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {

}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  shape="rounded"
  size="medium"
  type="button"
>
  Size
</button>
`;

exports[`test Button styles button size all render correct 3`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  padding: 12px 24px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {

}

@media (hover:none) {

}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  shape="rounded"
  size="large"
  type="button"
>
  Size
</button>
`;

exports[`test Button styles button size all render correct 4`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 16px;
  line-height: 24px;
  font-weight: 500;
  padding: 12px 24px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {

}

@media (hover:none) {

}

@media (hover:none) {

}

@media screen and (min-width:40em) {

}

@media screen and (min-width:52em) {
  .c0[type] {
    padding: 12px 24px;
  }
}

@media screen and (min-width:64em) {

}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  shape="rounded"
  size="response"
  type="button"
>
  Size
</button>
`;

exports[`test Button styles disabled button style 1`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  padding: 6px 16px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  disabled={true}
  shape="rounded"
  size="small"
  type="button"
>
  Disabled
</button>
`;

exports[`test Button styles loading button style 1`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  padding: 6px 16px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c2 {
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c5 {
  box-sizing: border-box;
  border-width: 4px;
  border-style: solid;
  border-radius: 50%;
  opacity: 0;
  -webkit-animation: dkykjR 1s infinite;
  animation: dkykjR 1s infinite;
  -webkit-animation-delay: .0s;
  animation-delay: .0s;
}

.c6 {
  box-sizing: border-box;
  border-width: 4px;
  border-style: solid;
  border-radius: 50%;
  opacity: 0;
  -webkit-animation: dkykjR 1s infinite;
  animation: dkykjR 1s infinite;
  -webkit-animation-delay: .1s;
  animation-delay: .1s;
}

.c7 {
  box-sizing: border-box;
  border-width: 4px;
  border-style: solid;
  border-radius: 50%;
  opacity: 0;
  -webkit-animation: dkykjR 1s infinite;
  animation: dkykjR 1s infinite;
  -webkit-animation-delay: .2s;
  animation-delay: .2s;
}

.c3 {
  width: 44px;
  line-height: 20px;
  padding: 6px 0;
}

.c3 .c4 {
  border-width: 4px;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 1;
  cursor: not-allowed;
}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  disabled={true}
  loading={true}
  shape="rounded"
  size="small"
  type="button"
>
  <div
    className="c2 c3"
  >
    <div
      className="c4 c5"
    />
    <div
      className="c4 c6"
    />
    <div
      className="c4 c7"
    />
  </div>
</button>
`;

exports[`test Button styles primary button style 1`] = `
.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 14px;
  line-height: 20px;
  font-weight: 500;
  padding: 6px 16px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<button
  className="c0 c1"
  shape="rounded"
  size="small"
  type="button"
>
  Primary
</button>
`;

exports[`test Button styles pureIcon button style 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  font-family: 'Ay-Web-Icons-Solid' !important;
}

.c3 {
  font-style: normal;
  font-weight: 400;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c3:before {
  content: '\\e922';
}

.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 16px;
  font-weight: 500;
  width: 32px;
  height: 32px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<body>
    <div>
      <button
        class="c0 c1"
        shape="rounded"
        type="button"
      >
        <span
          class="c2"
          type="solid"
        >
          <i
            class="c3"
          />
        </span>
        Loading
      </button>
    </div>
  </body>,
  "container": .c2 {
  font-family: 'Ay-Web-Icons-Solid' !important;
}

.c3 {
  font-style: normal;
  font-weight: 400;
  font-variant: normal;
  text-transform: none;
  line-height: 1;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

.c3:before {
  content: '\\e922';
}

.c0[type] {
  color: #ffffff;
  background-color: #1e9fd2;
  border-color: #1e9fd2;
  border-radius: 4px;
  font-size: 16px;
  font-weight: 500;
  width: 32px;
  height: 32px;
  border-width: 1px;
  border-style: solid;
}

.c0[type]:not(:disabled):hover {
  background-color: #0f7aab;
  border-color: #0f7aab;
}

.c0[type]:not(:disabled):active {
  background-color: #055885;
  border-color: #055885;
}

.c0[type]:not(:disabled):focus {
  border-color: #1e9fd2;
  box-shadow: 0 0 0 2px #1e9fd248;
}

.c1[type] {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
}

.c1[type]:disabled {
  opacity: 0.48;
  cursor: not-allowed;
}

@media (hover:none) {
  .c0[type]:hover {
    background-color: #1e9fd2;
  }
}

<div>
    <button
      class="c0 c1"
      shape="rounded"
      type="button"
    >
      <span
        class="c2"
        type="solid"
      >
        <i
          class="c3"
        />
      </span>
      Loading
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
